services/web/server/src/simcore_service_webserver/projects/_crud_api_create.py

```diff
@@ -286,6 +290,20 @@ async def create_project(  # pylint: disable=too-many-arguments  # noqa: C901, P
                 predefined_project=predefined_project,
             )
 
+        # If user wants to create project in specific workspace
+        if workspace_id:
+            # Verify user access to the specified workspace; raise an error if access is denied
+            workspace = await get_workspace(
```

Comment:
I think we really should have a check_workspace function. this is all copy-pasted everywhere. why not simplify this? and make the code more readable at the same time?